components/ws-manager-bridge/src/bridge.ts

```diff
@@ -285,11 +285,18 @@ export class WorkspaceManagerBridge implements Disposable {
                 // Do not override!
                 log.error(logContext, 'We received an empty "failed" condition overriding an existing one!', {
                     current: instance.status.conditions.failed,
+                    status,
                 });
-
-                // TODO(gpl) To make ensure we do not break anything big time we keep the unconditional override for now, and observe for some time.
-                instance.status.conditions.failed = status.conditions.failed;
             } else {
+                if (
+                    !!instance.status.conditions.failed &&
+                    instance.status.conditions.failed !== status.conditions.failed
+                ) {
+                    log.info(logContext, 'Overriding previous "failed" condition!', {
+                        previous: instance.status.conditions.failed,
+                        status,
+                    });
+                }
```

Comment:
Would you be able to provide a bit more context on why this is needed?

Normally in controller theory, the controller takes readings from the environment, reduces them to some data and updates the state with these. It should never look at previous events as that prevents the controller to self-correct and eventually converge to the correct solution.

Here, we're getting an update which says it's not failed. Why does the update contain a not-failed state after it reported a failed state? Did it correct itself? If so, why wouldn't we set our status to what we're getting from the observations?

Reply:
> Would you be able to provide a bit more context on why this is needed?

The original issue was that a workspace failed to start (initialize) and as a result didn't receive a backup. Usually, such workspaces have the failed condition set, so the workspace has an error indicator in the dashboard.

<img width="1342" alt="Screenshot 2022-08-25 at 14 09 53" src="https://user-images.githubusercontent.com/372735/186661070-1572c438-f370-4386-b4ae-3b884771ba22.png">

In this case ws-manager would send a status that overrides the `failed` condition, so the workspace looks ok and startable. When a user starts it they see

<img width="389" alt="Screenshot 2022-08-25 at 14 14 39" src="https://user-images.githubusercontent.com/372735/186661609-89ae9923-dee8-47cd-8c50-79aef744c549.png">

Thanks for helping me reflecting on this more and I agree with what you wrote. The core issue is that we don't have a clear signal of whether a workspace is broken (startable) or not. It's just less surprising/confusing when a workspace is already marked with an error. I'll think a bit more how we can better indicate this.

Reply:
x-refing: https://github.com/gitpod-io/gitpod/issues/12355#issuecomment-1229885538